webapp/src/main/java/org/apache/atlas/web/rest/TypeCacheRefreshREST.java

```diff
@@ -52,21 +52,25 @@ public TypeCacheRefreshREST(AtlasTypeDefStore typeDefStore, IAtlasGraphProvider
     @POST
     @Path("/refresh")
     @Timed
-    public void refreshCache(@QueryParam("expectedFieldKeys") int expectedFieldKeys) throws AtlasBaseException {
+    public void refreshCache(@QueryParam("expectedFieldKeys") int expectedFieldKeys, @QueryParam("traceId") String traceId) throws AtlasBaseException {
         try {
-            refreshTypeDef(expectedFieldKeys);
+            if (serviceState.getState() != ServiceState.ServiceStateValue.ACTIVE) {
+                LOG.warn("Node is in {} state. skipping refreshing type-def-cache :: traceId {}", serviceState.getState(), traceId);
+                return;
+            }
+            refreshTypeDef(expectedFieldKeys, traceId);
         } catch (Exception e) {
-            LOG.error("Error during refreshing cache " + e.getMessage(), e);
+            LOG.error("Error during refreshing cache  :: traceId " + traceId + " " + e.getMessage(), e);
```

Comment:
No need to log the error, instead throw a more detailed exception which contains e.getMessage(), e that we are logging